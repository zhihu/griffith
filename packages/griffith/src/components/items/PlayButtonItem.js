import React from 'react'
import {css} from 'aphrodite/no-important'
import styles from '../Controller.styles'
import Icon from '../Icon'
import * as icons from '../icons/controller'

const PlayButtonItem = ({isPlaying, onClick}) => (
  <div className={css(styles.toggle)}>
    <button className={css(styles.button)} onClick={onClick}>
      <Icon icon={isPlaying ? icons.pause : icons.play} />
    </button>
  </div>
)

export default PlayButtonItem
