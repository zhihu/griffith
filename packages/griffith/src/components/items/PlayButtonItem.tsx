import React from 'react'
import * as icons from '../icons/controller'
import ControllerButton from './ControllerButton'
import ControllerTooltip from './ControllerTooltip'

const PlayButtonItem: React.FC<{
  isPlaying: boolean
  onClick: React.HTMLAttributes<HTMLButtonElement>['onClick']
}> = ({isPlaying, onClick}) => (
  <ControllerTooltip content={isPlaying ? 'pause' : 'play'}>
    <ControllerButton
      icon={isPlaying ? icons.pause : icons.play}
      onClick={onClick}
    />
  </ControllerTooltip>
)

export default PlayButtonItem
