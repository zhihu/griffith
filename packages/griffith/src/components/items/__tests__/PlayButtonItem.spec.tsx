import React from 'react'
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'enzy... Remove this comment to see the full error message
import {shallow} from 'enzyme'
import PlayButtonItem from '../PlayButtonItem'

describe('PlayButtonItem', () => {
  it('get PlayButtonItem component', () => {
    const onClick = jest.fn()
    const wrapper = shallow(
      <PlayButtonItem isPlaying={true} onClick={onClick} />
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.find('button').simulate('click')
    expect(onClick).toBeCalled()
  })
})
