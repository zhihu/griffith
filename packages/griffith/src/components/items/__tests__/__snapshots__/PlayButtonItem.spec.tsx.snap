// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButtonItem get PlayButtonItem component 1`] = `
<div>
  <div class="menuContainer_e296pg">
    <button class="button_1j0k413" type="button">
      <span class="root_s9vkyf"
        ><svg viewBox="0 0 24 24" class="svg_11u3rwe">
          <path
            d="M9,5 C10,5 10,6 10,6 C10,6 10,17 10,18 C10,19 9,19 9,19 L7,19 C6,19 6,18 6,18 L6,6 C6,5 7,5 7,5 C7,5 8,5 9,5 Z M17,5 C18,5 18,6 18,6 C18,6 18,17 18,18 C18,19 17,19 17,19 L15,19 C14,19 14,18 14,18 L14,6 C14,5 15,5 15,5 C15,5 16,5 17,5 Z"
          ></path></svg
      ></span>
    </button>
    <div class="menu_1a7s4a3-o_O-tooltipContent_me14v4">Pause</div>
  </div>
</div>

`;
